pep-0695.rst

```diff
@@ -809,24 +813,78 @@ It also adds an AST node type that represents a type parameter.
         | ParamSpec(identifier name)
         | TypeVarTuple(identifier name)
 
+Bounds and constraints are represented identically in the AST. In the implementation,
+any expression that is syntactically a tuple is treated as a constraint, and any other
+expression is treated as a bound.
+
 It also modifies existing AST node types ``FunctionDef``, ``AsyncFunctionDef``
 and ``ClassDef`` to include an additional optional attribute called
-``typeparam*`` that includes a list of type parameters associated with the
+``typeparams`` that includes a list of type parameters associated with the
 function or class.
 
+Lazy Evaluation
+---------------
+
+This PEP introduces three new contexts where expressions may occur that represent
+static types: ``TypeVar`` bounds, ``TypeVar`` constraints, and the value of type
+aliases. As with type annotations, these expressions may contain references to names
+that are not yet defined. For example, type aliases may be recursive, or even mutually
```

Comment:
I found the "as with type annotations" bit somewhat confusing, since forward references in type annotations don't work at the moment at runtime, unless you have `from __future__ import annotations` at the top or the forward reference is in quotes.

Maybe something like this would be clearer?

```suggestion
aliases. As with type annotations, a type checker should not emit an error if these expressions contain references to names
that are not yet defined. For example, type aliases may be recursive, or even mutually
```

Reply:
I feel like that puts too much emphasis on type checkers. The primary purpose here is to make it easy for humans to understand and read the code, and we should design type checkers so that they understand the code we want.

Reply:
We could also just do this?

```suggestion
aliases. These expressions may contain references to names
that are not yet defined. For example, type aliases may be recursive, or even mutually
```

Reply:
Done!